src/simpledb/materialize/HashJoinScan.java

```diff
@@ -0,0 +1,149 @@
+package simpledb.materialize;
+
+import java.util.List;
+
+import simpledb.query.*;
+import simpledb.tx.Transaction;
+
+/**
+ * The Scan class for the <i>hashjoin</i> operator.
+ * 
+ * @author Edward Sciore
+ */
+public class HashJoinScan implements Scan {
+   private Transaction tx;
+   private List<TempTable> b1, b2;
+   private UpdateScan s1, s2;
+   private String fldname1, fldname2;
+   private int currIdx = 0;
+   private boolean hasJoinVal = false;
+
+   /**
+    * Create a hashjoin scan for the two underlying hash buckets.
+    * 
+    * @param b1       the LHS hashed buckets
+    * @param b2       the RHS hashed buckets
+    * @param fldname1 the LHS join field
+    * @param fldname2 the RHS join field
+    * @param tx       the current transaction
+    */
+   public HashJoinScan(Transaction tx, List<TempTable> b1, List<TempTable> b2, String fldname1, String fldname2) {
+      this.tx = tx;

```

Comment:
Transaction tx wasn't used in the functions so maybe can remove it from HashJoinScan constructor too